### PR TITLE
🐛Source Stripe: update endpoint for bank_accounts stream

### LINKED ISSUE #33714: Source Stripe: sources API deprecated
As part of the `bank_accounts` stream, we use the endpoint "customers/{customer_id}/sources". This endpoint is not available in the documentation and stripe mentions explicitly that the sources API is deprecated (see https://stripe.com/docs/sources/customers). We should move away from this API.

~Even though it's less clear that this is deprecated, the `expands=data.sources` parameter is not in the API doc anymore (I would like someone else to confirm that). It would mean that the lazy pattern we have right now might not work eventually and we would need to query all the child to see if there is a bank account.~ Based on the comment from @askarpets below, the previous statement is not true as the doc now has an [Expand page](https://stripe.com/docs/expand) stating how expanding works. For the customer object, `sources` is marked as expandable (see image) so the logic lazy logic should stay the same.

image

### PR DESCRIPTION
**Pull request by askarpets**
## What
Resolves https://github.com/airbytehq/airbyte/issues/33714

## How
Change `path` for the `bank_accounts` stream from `customers/{customer_id}/sources` to `customers/{customer_id}/bank_accounts`

## Recommended reading order
1. `source.py`
2. `bank_accounts.json`

## 🚨 User Impact 🚨
No breaking changes

## Pre-merge Actions

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-stripe/source_stripe/streams.py [hunk 1]
```diff
@@ -842,7 +842,7 @@ def parse_response(self, response: requests.Response, *args, **kwargs) -> Iterab
             # as the events API does not support expandable items. Parent class will try getting sub-items from this object,
             # then from its own API. In case there are no sub-items at all for this entity, API will raise 404 error.
             self.logger.warning(
-                "Data was not found for URL: {response.request.url}. "
+                f"Data was not found for URL: {response.request.url}. "
```
Comment (roman-yermilov-gl) [inline airbyte-integrations/connectors/source-stripe/source_stripe/streams.py:845]: 👍

**Review by roman-yermilov-gl — APPROVED** [review]

File: airbyte-integrations/connectors/source-stripe/integration_tests/expected_records.jsonl [hunk 2]
```diff
@@ -46,7 +46,7 @@
...
...on=sub_1O2Dg0EcXtiJtvvhz7Q4zS0n"}, "latest_invoice": "in_1OOKkUEcXtiJtvvheUUavyuB", "livemode": false, "metadata": {}, "next_pending_invoice_item_invoice": null, "on_behalf_of": null, "pause_collection": null, "payment_settings": {"payment_method_options": null, "payment_method_types": null, "save_default_payment_method": null}, "pending_invoice_item_interval": null, "pending_setup_intent": null, "pending_update": null, "plan": {"id": "price_1MSHZoEcXtiJtvvh6O8TYD8T", "object": "plan", "active": true, "aggregate_usage": null, "amount": 600, "amount_decimal": "600", "billing_scheme": "per_unit", "created": 1674209524, "currency": "usd", "interval": "month", "interval_count": 1, "livemode": false, "metadata": {}, "nickname": null, "product": "prod_NCgx1XP2IFQyKF", "tiers_mode": null, "transform_usage": null, "trial_period_days": null, "usage_type": "licensed"}, "quantity": 1, "schedule": "sub_sched_1O2Dg0EcXtiJtvvh7GtbtIhP", "start_date": 1697550676, "status": "active", "test_clock": null, "transfer_data": null, "trial_end": null, "trial_settings": {"end_behavior": {"missing_payment_method": "create_invoice"}}, "trial_start": null, "updated": 1697550676}, "emitted_at": 1700232971060}
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-stripe/integration_tests/expected_records.jsonl:49]: Should we ignore fields that are susceptible to changes like `current_period_start`, `current_period_end` and `latest_invoice`?

File: airbyte-integrations/connectors/source-stripe/integration_tests/expected_records.jsonl [hunk 3]
```diff
@@ -21,7 +21,7 @@
...
+{"stream": "customers", "data": {"id": "cus_NGoTFiJFVbSsvZ", "object": "customer", "address": {"city": "", "country": "US", "line1": "Street 2, 34567", "line2": "", "postal_code": "94114", "state": "CA"}, "balance": 0, "created": 1675160053, "currency": "usd", "default_source": "src_1MWGs8EcXtiJtvvh4nYdQvEr", "delinquent": false, "description": "Test Customer 2 description", "discount": null, "email": "user1.sample@zohomail.eu", "invoice_prefix": "C09C1837", "invoice_settings": {"custom_fields": null, "default_payment_method": null, "footer": null, "rendering_options": null}, "livemode": false, "metadata": {}, "name": "Test Customer 2", "next_invoice_sequence": 15, "phone": null, "preferred_locales": ["en-US"], "shipping": {"address": {"city": "", "country": "US", "line1": "Street 2, 34567", "line2": "", "postal_code": "94114", "state": "CA"}, "name": "Test Customer 2", "phone": ""}, "tax_exempt": "none", "test_clock": null, "updated": 1675160053}, "emitted_at": 1697627278439}
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-stripe/integration_tests/expected_records.jsonl:24]: Should we ignore fields that are susceptible to changes like `next_invoice_sequence `?

File: airbyte-integrations/connectors/source-stripe/integration_tests/expected_records.jsonl [hunk 4]
```diff
@@ -21,7 +21,7 @@
...ective_at": 1686158100, "invoice": "in_1K9GK0EcXtiJtvvhSo2LvGqT", "lines": {"object": "list", "data": [{"id": "cnli_1NGPwmEcXtiJtvvhcL7yEIBJ", "object": "credit_note_line_item", "amount": 8400, "amount_excluding_tax": 8400, "description": "a box of parsnips", "discount_amount": 0, "discount_amounts": [], "invoice_line_item": "il_1K9GKLEcXtiJtvvhhHaYMebN", "livemode": false, "quantity": 1, "tax_amounts": [], "tax_rates": [], "type": "invoice_line_item", "unit_amount": 8400, "unit_amount_decimal": 8400.0, "unit_amount_excluding_tax": 8400.0}], "has_more": false, "url": "/v1/credit_notes/cn_1NGPwmEcXtiJtvvhNXwHpgJF/lines"}, "livemode": false, "memo": null, "metadata": {}, "number": "CA35DF83-0001-CN-01", "out_of_band_amount": null, "pdf": "https://pay.stripe.com/credit_notes/acct_1JwnoiEcXtiJtvvh/test_YWNjdF8xSndub2lFY1h0aUp0dnZoLF9PMlV3dFlJelh4NHM1R0VIWnhMR3RjWUtlejFlRWtILDg4MTY4MDc20200Sa50llWu/pdf?s=ap", "reason": null, "refund": null, "shipping_cost": null, "status": "issued", "subtotal": 8400, "subtotal_excluding_tax": 8400, "tax_amounts": [], "total": 8400, "total_excluding_tax": 8400, "type": "pre_payment", "voided_at": null, "updated": 1686158100}, "emitted_at": 1697627276386}
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-stripe/integration_tests/expected_records.jsonl:21]: Does this mean that [this](https://airbytehq-team.slack.com/archives/C02U9R3AF37/p1704383358320959) has been fixed?

**Review by maxi297 — APPROVED** [review]
LGTM! Just a couple of questions for my personal knowledge and to maybe improve maintainability. Thanks @askarpets !

File: airbyte-integrations/connectors/source-stripe/integration_tests/expected_records.jsonl (hunk 4 above)
Comment (askarpets) [reply airbyte-integrations/connectors/source-stripe/integration_tests/expected_records.jsonl:21]: Sorry, I don't have access to this link. Could you please describe the content here?

File: airbyte-integrations/connectors/source-stripe/integration_tests/expected_records.jsonl (hunk 2 above)
Comment (askarpets) [reply airbyte-integrations/connectors/source-stripe/integration_tests/expected_records.jsonl:49]: I see that these fields were previously changed 2 months ago. Do you think it is too frequent so we should add it to ignore fields list? I'm just not sure if we should ignore them in such cases.

File: airbyte-integrations/connectors/source-stripe/integration_tests/expected_records.jsonl (hunk 3 above)
Comment (askarpets) [reply airbyte-integrations/connectors/source-stripe/integration_tests/expected_records.jsonl:24]: This change is related to the previous one based on customer id field so should be ignored if we decide to ignore the previous case's fields.

File: airbyte-integrations/connectors/source-stripe/integration_tests/expected_records.jsonl (hunk 2 above)
Comment (maxi297) [reply airbyte-integrations/connectors/source-stripe/integration_tests/expected_records.jsonl:49]: What is the value in asserting the value of those fields specifically? I don't have context to answer that question but it feels like it's smaller than the engineering time to keep this updated

File: airbyte-integrations/connectors/source-stripe/integration_tests/expected_records.jsonl (hunk 4 above)
Comment (maxi297) [reply airbyte-integrations/connectors/source-stripe/integration_tests/expected_records.jsonl:21]: Discussed synchronously: it seems like the issue was resolved but we are clueless about how. We will leave it at that for now

File: airbyte-integrations/connectors/source-stripe/integration_tests/expected_records.jsonl (hunk 2 above)
Comment (askarpets) [reply airbyte-integrations/connectors/source-stripe/integration_tests/expected_records.jsonl:49]: Marked the fields as ignored

File: airbyte-integrations/connectors/source-stripe/integration_tests/expected_records.jsonl (hunk 3 above)
Comment (askarpets) [reply airbyte-integrations/connectors/source-stripe/integration_tests/expected_records.jsonl:24]: Marked the field as ignored

**Pull request merged**